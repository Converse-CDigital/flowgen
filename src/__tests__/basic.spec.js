// @flow

import { compiler, beautify } from "..";

it("should handle basic keywords", () => {
  const ts = `type A = {
    a: void,
    b: string,
    c: any,
    d: number,
    e: boolean,
    f: null,
    g: undefined,
    h: object,
    i: 1,
    j: 2,
    k: true,
    l: false,
    m: 'foo',
    n: 'bar',
    o: never,
    p: unknown,
    r: -1,
    s: -2,
    t: symbol,
    u: unique symbol,
  }`;
  const result = compiler.compileDefinitionString(ts, {quiet: true});
  expect(beautify(result)).toMatchSnapshot();
});
