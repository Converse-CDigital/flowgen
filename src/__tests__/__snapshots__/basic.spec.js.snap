// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle basic keywords 1`] = `
"declare type A = {
  a: void,
  b: string,
  c: any,
  d: number,
  e: boolean,
  f: null,
  g: void,
  h: { [key: string]: any },
  i: 1,
  j: 2,
  k: true,
  l: false,
  m: \\"foo\\",
  n: \\"bar\\",
  o: empty,
  p: mixed,
  r: -1,
  s: -2,
  t: Symbol,
  u: Symbol,
  v: [1, 2, 3],
  w: $ReadOnlyArray<string>,
  ...
};
"
`;
